### PR TITLE
Add synchronized queues

### PR DESCRIPTION
**Pull request by CodingRays**
We will need to synchronize access to our queues once we start adding asynchronous commits. Luckily there are only 4 functions that need synchronized access so i think a small wrapper for queue objects with a lock should suffice. Its simple and stable.

### CODE REVIEW

**Comment by hYdos** [comment]
sounds good!

File: Rosella/src/main/java/me/hydos/rosella/device/VulkanQueue.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    public int getQueueFamily() {
+        return this.family;
+    }
+
+    public int vkQueueSubmit(VkSubmitInfo submit, long fence) {
+        int result;
+        try {
+            this.lock.lock();
```
Comment (hYdos) [inline Rosella/src/main/java/me/hydos/rosella/device/VulkanQueue.java:35]: should we make a function called lock() ? just a suggestion, nothing too important

Comment (OroArmor) [reply Rosella/src/main/java/me/hydos/rosella/device/VulkanQueue.java:35]: and maybe an unlock as well

File: Rosella/src/main/java/me/hydos/rosella/device/VulkanQueue.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            result = VK10.vkQueueBindSparse(this.queue, pBindInfo, fence);
+        } finally {
+            this.lock.unlock();
+        }
+        return result;
+    }
+
+    // THIS IS BAD. VERY VERY BAD. DO NOT DO THIS. EVER... (but if for some reason you do need to here is a safe function)
```
Comment (hYdos) [inline Rosella/src/main/java/me/hydos/rosella/device/VulkanQueue.java:76]: use javadoc instead of comments
javadoc examples are in the other classes

**Review by hYdos — CHANGES_REQUESTED** [review]

File: Rosella/src/main/java/me/hydos/rosella/device/VulkanQueue.java (hunk 1 above)
Comment (CodingRays) [reply Rosella/src/main/java/me/hydos/rosella/device/VulkanQueue.java]: Done

File: Rosella/src/main/java/me/hydos/rosella/device/VulkanQueue.java (hunk 2 above)
Comment (CodingRays) [reply Rosella/src/main/java/me/hydos/rosella/device/VulkanQueue.java:76]: I used a comment since it wasnt really a documentation of the function

Comment (OroArmor) [reply Rosella/src/main/java/me/hydos/rosella/device/VulkanQueue.java:76]: use `@Deprecated` on it

Comment (CodingRays) [reply Rosella/src/main/java/me/hydos/rosella/device/VulkanQueue.java]: Done

**Review by OroArmor — APPROVED** [review]

**Review by hYdos — APPROVED** [review]

**Pull request merged**